cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/OAuth20RevokeClientTokensController.java

```diff
@@ -68,11 +74,29 @@ protected ModelAndView handleRequestInternal(final HttpServletRequest request, f
             return OAuthUtils.writeJsonError(response, OAuthConstants.INVALID_REQUEST, e.getMessage(), HttpStatus.SC_BAD_REQUEST);
         }
 
-        if (!centralOAuthService.revokeClientTokens(clientId, clientSecret)) {
+        // Verify that the client service has a valid client_id and client_secret
+        final OAuthRegisteredService service = centralOAuthService.getRegisteredService(clientId);
+        if (service == null || !service.getClientSecret().equals(clientSecret)) {
             LOGGER.error("Could not revoke client tokens, mismatched client id or client secret");
-            return OAuthUtils.writeJsonError(response, OAuthConstants.INVALID_REQUEST,
+            return OAuthUtils.writeJsonError(
+                    response,
+                    OAuthConstants.INVALID_REQUEST,
                     OAuthConstants.INVALID_CLIENT_ID_OR_SECRET_DESCRIPTION,
-                    HttpStatus.SC_BAD_REQUEST);
+                    HttpStatus.SC_BAD_REQUEST
+            );
+        }
+
+        // Retrieve the tokens from the registry and remove all of them
+        TokenRegistry tokenRegistry = centralOAuthService.getTokenRegistry();
```

Comment:
Use the exposed `tokenRegistry` to get a list of all tokens and remove them using `for` loops.
No more bad `while` loops.